addons/mail/models/res_partner.py

```diff
@@ -261,13 +261,15 @@ def _get_mention_suggestions_domain(self, search):
         ])
 
     @api.model
-    def _search_mention_suggestions(self, domain, limit):
+    def _search_mention_suggestions(self, domain, limit, access_domain=None):
         domain_is_user = expression.AND([[('user_ids', '!=', False)], [('user_ids.active', '=', True)], domain])
         priority_conditions = [
             expression.AND([domain_is_user, [('partner_share', '=', False)]]),  # Search partners that are internal users
             domain_is_user,  # Search partners that are users
             domain,  # Search partners that are not users
         ]
+        if access_domain:
+            priority_conditions[0] = access_domain
```

Comment:
This will break the order the results are fetched. Because of this it will first fetch all the internal users even if they are not members of the channel. While, you would expect to get the member users first. So, this domain should be the last one to be queried.